Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -193,6 +215,94 @@ HexEditorWidget::HexEditorWidget()
+    for (int i = 0; i < 8; ++i) {
+        Optional<u8> read_result = m_editor->get_byte(position + i);
+        u8 current_byte = 0;
+        if (!read_result.has_value())
+            break;
+
+        current_byte = read_result.release_value();
+        if (m_value_inspector_little_endian)
+            unsigned_long = ((u64)current_byte << (8 * byte_read_count)) + unsigned_long;
+        else
+            unsigned_long = (unsigned_long << 8) + current_byte;
+
+        ++byte_read_count;
+    }
+
+    if (!m_value_inspector_little_endian) {
+        /* if we didn't read far enough, lets finish shifting the bytes so the code below works */
+        size_t bytes_left_to_read = 8 - byte_read_count;
+        unsigned_long = (unsigned_long << (8 * bytes_left_to_read));
+    }
+
+    /* Populate the model */
+    ValueInspectorModel* m_value_inspector_model = new ValueInspectorModel();
+    if (byte_read_count >= 1) {
+        u8 unsigned_byte_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_byte_value = (unsigned_long & 0xFF);
+        else
+            unsigned_byte_value = (unsigned_long >> (64 - 8));
+
+        m_value_inspector_model->set_parsed_value(InspectorValueType::SignedByte, String::formatted("{}", (i8)unsigned_byte_value));
+        m_value_inspector_model->set_parsed_value(InspectorValueType::UnsignedByte, String::formatted("{}", unsigned_byte_value));
+    } else {
+        m_value_inspector_model->set_parsed_value(InspectorValueType::SignedByte, "");
+        m_value_inspector_model->set_parsed_value(InspectorValueType::UnsignedByte, "");
+    }
+
+    if (byte_read_count >= 2) {
+        u16 unsigned_short_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_short_value = (unsigned_long & 0xFFFF);
+        else
+            unsigned_short_value = (unsigned_long >> (64 - 16));
+
+        m_value_inspector_model->set_parsed_value(InspectorValueType::SignedShort, String::formatted("{}", (i16)unsigned_short_value));
+        m_value_inspector_model->set_parsed_value(InspectorValueType::UnsignedShort, String::formatted("{}", unsigned_short_value));
+    } else {
+        m_value_inspector_model->set_parsed_value(InspectorValueType::SignedShort, "");
+        m_value_inspector_model->set_parsed_value(InspectorValueType::UnsignedShort, "");
+    }
+
+    if (byte_read_count >= 4) {
+        u16 unsigned_int_value = 0;
+        if (m_value_inspector_little_endian)
+            unsigned_int_value = (unsigned_long & 0xFFFFFFFF);
+        else
+            unsigned_int_value = (unsigned_long >> 32);
+
+        m_value_inspector_model->set_parsed_value(InspectorValueType::SignedInt, String::formatted("{}", (i32)unsigned_int_value));
+        m_value_inspector_model->set_parsed_value(InspectorValueType::UnsignedInt, String::formatted("{}", unsigned_int_value));
+        m_value_inspector_model->set_parsed_value(InspectorValueType::Float, String::formatted("{}", (float)unsigned_int_value));
```

Comment:
I'm surprised that C-cast works. I think bit_cast would be more intentional and clearer.

Reply:
A C-cast would not work to interpret the `unsigned_int_value` bits directly as a float, it would convert to a float with the same value as `unsigned_int_value` 

(You can see in the screenshot the float value is just the unsigned int value + some error)

Likewise for double

Reply:
Thanks for the explanation here and good catch noticing the effect on the screenshot (something I obviously didn't notice). This has been changed to a bit_cast in the most recent push.